logr.go

```diff
@@ -352,18 +359,33 @@ func (l Logger) Info(msg string, keysAndValues ...interface{}) {
 // triggered this log line, if present. The err parameter is optional
 // and nil may be passed instead of an error instance.
 func (l Logger) Error(err error, msg string, keysAndValues ...interface{}) {
-	if l.sink == nil {
+	if l.sink == nil || l.sink == discardLogSinkInstance {
```

Comment:
should we make the same short0circuit in Info() ?

Reply:
`DiscardLogSink.Enabled()` always returns false, so it didn't seem necessary. I've added it anyway, for the sake of consistency. I also added a `Logger.skipSink` helper function for clarity - at least I hope the name makes it clearer...

Reply:
l.Info() calls l.Enabled() calls l.Sink.Enabled().  This crosses the interface boundary, so possible optimizations are defeated.  I think checking it explicitly makes sense.

You say "I've added it anyway" but I don't see it here?

Reply:
I haven't pushed my latest state yet. I was still working on performance comparisons yesterday.